### PR TITLE
Create a new address implementation

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR adds the Address creation flow (handler + front page with forms). I added the related integration test. 

To avoid duplicated code, I reworked a bit the `update-user-address-form.tsx` to be usable for both update and creation of an address. To do so, it now takes a boolean indicating if it's a creation flow or an update one. From this boolean, we can adapting display and form behaviour for both flows.

I also did a little rework of some handlers, to follow what's had been done lastly with changes on API routes HTTP verbs handling.

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Screenshots

image

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-jk3wz3 Create new UserAddress with OpenAPI client and mock server](https://app.clickup.com/t/jk3wz3)

File: src/components/forms/profile/user-address-form.tsx [hunk 1]
```diff
@@ -43,31 +44,46 @@ const UpdateUserAddressForm: React.FC<{
...
+                  return;
+                }
 
-            throw new Error("Something went wrong");
-          });
+                throw new Error("Something went wrong");
+              },
+            );
+          } else {
+            await fetchJson(
```
Comment (Fuzznimp) [inline src/components/forms/profile/user-address-form.tsx:61]: Not sure we need the `else` here, don't we ?

File: tests/pages/add-address-page.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import userEvent from "@testing-library/user-event";
```
Comment (Fuzznimp) [inline tests/pages/add-address-page.tsx:1]: You forgot to add `.test` in your name file :).

Comment (Oluane) [reply tests/pages/add-address-page.tsx:1]: 🤦

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**